server/src/main/scala/io/delta/sharing/server/model.scala

```diff
@@ -135,3 +135,126 @@ object Action {
   // Basically delta sharing doesn't support write for now.
   val maxWriterVersion = 0
 }
+
+/**
+ * Actions defined in delta format, used in response when requested format is delta.
+ */
+
+sealed trait DeltaAction {
+  /** Turn this object to the [[DeltaSingleAction]] wrap object. */
+  def wrap: DeltaSingleAction
+}
+
+/**
+ * Used to block older clients from reading the shared table when backwards
+ * incompatible changes are made to the protocol. Readers and writers are
+ * responsible for checking that they meet the minimum versions before performing
+ * any other operations.
+ *
+ * Since this action allows us to explicitly block older clients in the case of a
+ * breaking change to the protocol, clients should be tolerant of messages and
+ * fields that they do not understand.
+ */
+case class DeltaProtocol(minReaderVersion: Int) extends DeltaAction {
+  override def wrap: DeltaSingleAction = DeltaSingleAction(protocol = this)
+}
+
+/**
+ * DeltaAddFile used in delta sharing protocol, copied from AddFile in delta.
+ *   Adding 4 fields: id/version/timestamp/expirationTimestamp.
+ *   Ignoring 1 field: tags.
```

Comment:
Question: are these fields all require minReaderVersion > 2?
If so we would skip them in the oss server change for now, since this change focuses on responseFormat=delta, we could add another PR to support DV, etc.